pkg/plugins/gateway/algorithms/prefix_cache.go

```diff
@@ -18,113 +18,205 @@ package routingalgorithms
 
 import (
 	"fmt"
+	"math"
 	"math/rand"
+	"sort"
 	"strconv"
 
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer"
-	"github.com/vllm-project/aibrix/pkg/plugins/gateway/prefixcacheindexer/tokenizer"
+	"github.com/vllm-project/aibrix/pkg/cache"
+	"github.com/vllm-project/aibrix/pkg/plugins/gateway/algorithms/prefixcacheindexer"
```

Comment:
The current naming feels a bit unclear. Previously, we used prefixcacheindexer to refer to the component handling indexing. 

I think we need better structure here.

Option 1: We can have a folder for prefixcache and move files under `prefixcacheindexer` directly under it or just move the `indexer` folder under it.  

```
gateway/
├── algorithms/
│   └── prefixcache/
│       ├── indexer/
```

Option 2: Keep it as original folder, the algorithm folder still leverage outside util to extend the algorithm part. 

```
gateway/
├── algorithms/
 │       ├── prefix_cache.go
 │       ├── prefix_cache_and_load.go
 │       ├── prefix_cache_distributed_hash.go
```
\

Reply:
For algorithms, we should keep them as option 2.
```
gateway/
├── algorithms/
 │       ├── prefix_cache.go
 │       ├── prefix_cache_and_load.go
 │       ├── prefix_cache_distributed_hash.go
```

We can refactor prefixcacheindexer which continues to have hash & tree based implementation which are specific to prefix_cache usecase and move prefixcacheindex/cache/lru_store files to util package for any future usecase. Final outcome will look like this

```
gateway/
├── algorithms/
 |       ├── prefixcacheindexer
 |                |── hash.go
 |                |── tree.go           
 │       ├── prefix_cache.go
 │       ├── prefix_cache_and_load.go
 │       ├── prefix_cache_distributed_hash.go

utils
├──lrustore
│              |──lru_store.go            
```

Reply:
sounds good to me. Let's move them to utils